pkg/groups/validation.go

```diff
@@ -35,7 +41,8 @@ func (p *PayloadValidator) Validate(ctx context.Context, method, endpoint string
 	if p.isCreateGroup(method, endpoint) {
```

Comment:
Just a **NONBLOCKING** comment: (understand this is out of this PR's scope.)

These `if` blocks smell a bit risky if some endpoints are added or deleted (open/closed principle). I guess it will be improved in the future?

Reply:
I agree, not sure if there is a plan for that (I also don't like how it's a bunch of `if`s, but in reality it is more like a `switch` or `if/else` statement)

Reply:
Unfortunately there is no way to generalize the validation of the payload from the controller logic that needs to handle it, so when a new endpoint gets added then necessarily you need to add validation for it, otherwise you just can't validate that payload
Also, notice that same types are validated differently based on which handlers they are for (for example POST Group or PUT Group, different validation).

`if`s are usually not nice to look at but branching in this case is more dynamic and based on more than one value, and the alternative would be to hide the "ifs" with an OOP approach, that's why this was kept as simple as possible.

I'm sure we'll focus on improving things after deadlines are met.

Reply:
@wood-push-melon open an issue with your suggestion so we can track it and keep it in mind for improvements!

Reply:
gotcha. will do. thanks.